modules/silverpushBidAdapter.js

```diff
@@ -0,0 +1,321 @@
+  }
+};
+
+registerBidder(spec);
+
+export const CONVERTER = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 300
+  },
+  imp(buildImp, bidRequest, context) {
+    let imp = buildImp(bidRequest, context);
+
+    if (bidRequest.mediaTypes[VIDEO]) {
+      imp = buildVideoImp(bidRequest, imp);
+    } else if (bidRequest.mediaTypes[BANNER]) {
+      imp = buildBannerImp(bidRequest, imp);
+    }
+
+    const bidFloor = getBidFloor(bidRequest);
+
+    utils.deepSetValue(imp, 'bidfloor', bidFloor);
+
+    if (bidRequest.params.deals && bidRequest.params.deals.length > 0) {
+      utils.deepSetValue(imp, 'pmp', { deals: bidRequest.params.deals });
+    }
+
+    return imp;
+  },
+  request(buildRequest, imps, bidderRequest, context) {
+    const req = buildRequest(imps, bidderRequest, context);
+    mergeDeep(req, {
+      at: 1,
+      ext: {
+        bc: `${bidderConfig}_${bidderVersion}`
+      }
+    })
+
+    let userAgent = navigator.userAgent;
+    utils.deepSetValue(req, 'device.os', spec.getOS(userAgent));
+    utils.deepSetValue(req, 'device.devicetype', _isMobile() ? 1 : _isConnectedTV() ? 3 : 2);
+
+    const bid = context.bidRequests[0];
+    if (bid.params.publisherId) {
+      utils.deepSetValue(req, 'site.publisher.id', bid.params.publisherId);
+    }
+
+    return req;
+  },
+
+  bidResponse(buildBidResponse, bid, context) {
+    let bidResponse = buildBidResponse(bid, context);
+
+    if (bid.ext) {
+      bidResponse.meta.networkId = bid.ext.dsp_id;
+      bidResponse.meta.advertiserId = bid.ext.buyer_id;
+      bidResponse.meta.brandId = bid.ext.brand_id;
+    }
+
+    if (context.ortbResponse.ext && context.ortbResponse.ext.paf) {
+      bidResponse.meta.paf = Object.assign({}, context.ortbResponse.ext.paf);
+      bidResponse.meta.paf.content_id = utils.deepAccess(bid, 'ext.paf.content_id');
+    }
+
+    bidResponse = buildVideoVastRespose(bidResponse)
+    bidResponse = buildVideoOutstreamResponse(bidResponse, context)
+
+    return bidResponse;
+  },
+  response(buildResponse, bidResponses, ortbResponse, context) {
+    const response = buildResponse(bidResponses, ortbResponse, context);
+
+    let fledgeAuctionConfigs = utils.deepAccess(ortbResponse, 'ext.fledge_auction_configs');
+    if (fledgeAuctionConfigs) {
+      fledgeAuctionConfigs = Object.entries(fledgeAuctionConfigs).map(([bidId, cfg]) => {
+        return Object.assign({
+          bidId,
+          auctionSignals: {}
+        }, cfg);
+      });
+      return {
+        bids: response.bids,
+        fledgeAuctionConfigs,
+      }
+    } else {
+      return response.bids
+    }
+  }
+});
+
+function isBidRequestValid(bidRequest) {
+  return (utils.deepAccess(bidRequest, 'params.publisherId') != null && (isValidBannerRequest(bidRequest) || isValidVideoRequest(bidRequest)));
```

Comment:
According to your documents, publisherId should be a string. However, in this validation, it can also be a number or an empty string.

Reply:
Missed a few checks, changes done. 😄